### DISCUSSION TITLE
EPG Tool : timezone

### DISCUSSION THREAD

**Discussion by oliv615** [iptv-org/community#565] [Q&A]
Hello,
How to set the time zone in the EPG tool? I would like to have the Europe/Paris region. 
Thanks.

**Comment by IntProst** [comment]
I have a similar problem.  My XML file gets created showing entries like:
`<programme start="20240521043000 +0000" {etc.}`
but I need it to be:
`<programme start="20240521043000 -0400"  {etc.}`

My Windows 10 pc shows "Eastern Standard Time" when I type `tzutil /g` into CMD (command window), so it should be setting the `-0400`.  I'm not sure what I'm doing wrong.

**Comment by Sphinxroot** [comment]
it depend  on what you use to generate the epg file   , if your epg file is 12hr or 24hr or more ,   u may use  tvg-shift="" EPG Calibration if in foreign location to fit your timezone ( "+1" "-1" = Hour )   Empty=Normal  in your m3u playlist  for  the selected channel

#EXTINF:-1  tvg-id="" tvg-name="" tvg-shift="" tvg-country="" tvg-language="" tvg-chno="" tvg-logo="" tvg-logo-small=""  group-title="",CHANNEL NAME (720p/1080p)

may or may not work on your player

**Comment by IntProst** [comment]
[Sphinxroot](https://github.com/Sphinxroot),

Thanks for your reply.  I'm using the [EPG grabber](https://github.com/iptv-org/epg) from this site.  I'm not sure how to make it add the `tvg-shift=""` code to my resulting _Guide.xml_.  It would certainly be easier if I could.

I should mention as well that I'm using _MediaPortal_ with it's XMLTV import script to process the EPG.  The XMLTV script does provide a way to add or subtract time from the EPG xml file, but it just does not want to work for me.  I've tried several times to get it working, including reboots to make sure settings were saved, but I had no success.

However, I did find a way to edit the _Guide.xml_ file using a Windows' _PowerShell_ command.  I tried to create a single .BAT file to run using _Task Scheduler_, but oddly, any commands in the .BAT file after the [EPG grabber](https://github.com/iptv-org/epg) were skipped.  As a result, I needed to create two .BAT files to run sequentially in _Task Scheduler_.

First is my _GetEPG.BAT_:
```shell
npm run grab -- --days=14 --maxConnections=7 --channels=channel.list/TVGuide.xml --output=Guide.xml

``` 

Then I run my _TimeZone.BAT_:
```shell
powershell -Command "& {(gc Guide.xml) -replace '\+0000"', '-0400"' | sc TVGuide.xml}"
``` 
The file I use for _MediaPortal_ is the `sc` (output) file named _TVGuide.xml_.  Note that I need to use the [Escaping Character](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_regular_expressions?view=powershell-7.4#escaping-characters) pair of `\+` to search for the plus sign.   _PowerShell_ needs this because the plus sign is a reserved character.  I presume you would need the same for the replacement if your time zone is "plus" from Greenwich Mean Time (GMT).

My next task will be to modify the _TimeZone.BAT_ to adjust for Eastern Daylight Savings Time (-0500 from GMT).

I don't think I'll bother, but I could apply a variation of the _PowerShell_ solution to add `tvg-shift="-0400"` in the header of _TVGuide.xml_.
